### ISSUE TITLE
[A11y] Value returned for `announceListItemNumbering` in `announcerStringGetter` is not used when numbered list is created with `toggleNumbering` api

### ISSUE THREAD

**Issue by vhuseinova-msft** [microsoft/roosterjs#2734]
**Describe the bug**
RoosterJS provides a way to set a custom announcements for new items in numbered list (`announcerStringGetter`). But it looks like value returned for `announceListItemNumbering` in `announcerStringGetter` is not used when numbered list is created with `toggleNumbering` api. 
Please note that numbered lists that are created with auto formatting, use `announceListItemNumbering` as expected.

**To Reproduce**
1. Open Rooster JS demo web site
2. Click add numbered list button
3. type some text
4. Press enter
5. Observe that Narrator/VoiceOver doesn't announce a custom announcement for  `announceListItemNumbering`

**Expected behavior**
The `announceListItemNumbering` string returned in `announcerStringGetter` should be used to announce when a new bullet is added (in autoformatted list and when list is added with `toggleNumbering` api)

**Device Information**
 - OS: [e.g. iOS] MacOS
 - Browser [e.g. chrome, safari] Edge

**Comment by vhuseinova-msft** [comment]
cc @BryanValverdeU
